aiocouch/remote.py

```diff
@@ -256,6 +269,71 @@ def endpoint(self):
         )
 
 
+class RemoteAttachment(object):
+    def __init__(self, document, id):
+        self._document = document
+        self.id = id
+        self.content_type = None
+
+    @property
+    def endpoint(self):
+        return f"{self._document.endpoint}/{_quote_id(self.id)}"
+
+    @raises(401, "Read privilege required for document '{document_id}'")
+    @raises(403, "Read privilege required for document '{document_id}'")
+    async def _exists(self):
+        try:
+            resp, data = await self._document._database._remote._head(
+                self.endpoint, return_response=True
+            )
+            self.content_type = resp.headers["Content-Type"]
+            return True
+        except aiohttp.ClientResponseError as e:
+            if e.status == 404:
+                return False
+            else:
+                raise e
+
+    @raises(400, "Invalid request parameters")
+    @raises(401, "Read privilege required for document '{document_id}'")
+    @raises(403, "Read privilege required for document '{document_id}'")
+    @raises(404, "Document '{document_id}' or attachment '{id}' doesn’t exists")
+    async def _get(self, **params):
+        resp, data = await self._document._database._remote._get(
+            self.endpoint, params, return_response=True
+        )
+        self.content_type = resp.headers["Content-Type"]
+        return data
+
+    @raises(400, "Invalid request body or parameters")
+    @raises(401, "Write privilege required for document '{document_id}'")
+    @raises(403, "Write privilege required for document '{document_id}'")
+    @raises(404, "Document '{document_id}' doesn’t exists")
+    @raises(
+        409, "Specified revision {document_rev} is not the latest for target document"
+    )
+    async def _put(self, rev, data, content_type, **params):
+        params["rev"] = rev
+        data = await self._document._database._remote._put(
+            self.endpoint, data, params, content_type
+        )
+        self.content_type = content_type
+        return data
```

Comment:
This line may seem useless, it's in case some caller (from within aiocouch (in the future) or from user code) needs the info returned by CouchDB. Same for `_delete()`.